chain/solana/src/chain.rs

```diff
@@ -0,0 +1,299 @@
+    ) -> Result<Box<dyn BlockStream<Self>>, Error> {
+        panic!("SOLANA does not support polling block stream")
+    }
+
+    fn chain_store(&self) -> Arc<dyn ChainStore> {
+        self.chain_store.clone()
+    }
+
+    async fn block_pointer_from_number(
+        &self,
+        _logger: &Logger,
+        _number: BlockNumber,
+    ) -> Result<BlockPtr, IngestorError> {
+        // FIXME (Solana): Hmmm, what to do with this?
+        Ok(BlockPtr {
+            hash: BlockHash::from(vec![0xff; 32]),
+            number: 0,
+        })
+    }
+
+    fn runtime_adapter(&self) -> Arc<Self::RuntimeAdapter> {
+        Arc::new(RuntimeAdapter {})
+    }
+
+    fn is_firehose_supported(&self) -> bool {
+        true
+    }
+}
+
+pub struct TriggersAdapter {}
+
+#[async_trait]
+impl TriggersAdapterTrait<Chain> for TriggersAdapter {
+    fn ancestor_block(
+        &self,
+        _ptr: BlockPtr,
+        _offset: BlockNumber,
+    ) -> Result<Option<codec::Block>, Error> {
+        // FIXME (Solana):  Might not be necessary for Solana support for now
+        Ok(None)
+    }
+
+    async fn scan_triggers(
+        &self,
+        _from: BlockNumber,
+        _to: BlockNumber,
+        _filter: &TriggerFilter,
+    ) -> Result<Vec<BlockWithTriggers<Chain>>, Error> {
+        // FIXME (Solana): Scanning triggers makes little sense in Firehose approach, let's see
+        Ok(vec![])
+    }
+
+    async fn triggers_in_block(
+        &self,
+        _logger: &Logger,
+        block: codec::Block,
+        _filter: &TriggerFilter,
+    ) -> Result<BlockWithTriggers<Chain>, Error> {
+        let shared_block = Arc::new(block.clone());
+        let instructions = block.transactions.iter().flat_map(|transaction| {
+            //let transaction_id = transaction.id.clone();
+            let b = shared_block.clone();
+            let tx = transaction.clone();
+            transaction.instructions.iter().flat_map(move |instruction| {
+                Some(trigger::InstructionWithInfo {
+                    instruction: instruction.clone(),
+                    block_num: b.number,
+                    block_id: b.id.clone(),
+                    transaction_id: tx.id.clone(),
+                })
+            })
+        });
+
+        let mut trigger_data: Vec<_> = instructions
+            .map(|i| SolanaTrigger::Instruction(Arc::new(i)))
+            .collect();
+        trigger_data.push(SolanaTrigger::Block(shared_block.cheap_clone()));
+
+        Ok(BlockWithTriggers::new(block, trigger_data))
+    }
+
+    async fn is_on_main_chain(&self, _ptr: BlockPtr) -> Result<bool, Error> {
+        // FIXME (Solana): Might not be necessary for Solana support for now
+        Ok(true)
+    }
+
+    /// Panics if `block` is genesis.
+    /// But that's ok since this is only called when reverting `block`.
+    async fn parent_ptr(&self, block: &BlockPtr) -> Result<Option<BlockPtr>, Error> {
+        // FIXME (NEAR):  Might not be necessary for NEAR support for now
```

Comment:
Needs to be implemented correctly now since used on reverts to go back in "time" properly, check NEAR implementation.

Reply:
this is the same implementation as on NEAR.  unless there's a different implementation on another branch?

Reply:
No I mixed it up and thought it was another one, needs to be a `panic!("Should never be called since FirehoseBlockStream cannot resolve it")` (I'll need to update NEAR version later also)